staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers_test.go

```diff
@@ -471,22 +497,99 @@ func TestDeferredResponseWriter_Write(t *testing.T) {
 				if !bytes.Equal(fullPayload, decompressed) {
 					t.Errorf("payload mismatch, expected: %s, got: %s", fullPayload, decompressed)
 				}
-
 			} else {
-				if contentEncoding != "" {
-					t.Errorf("content-encoding is set unexpectedly")
-				}
-
-				if strings.Contains(varyHeader, "Accept-Encoding") {
-					t.Errorf("accept encoding is set unexpectedly")
-				}
-
 				if !bytes.Equal(fullPayload, resBytes) {
 					t.Errorf("payload mismatch, expected: %s, got: %s", fullPayload, resBytes)
 				}
-
 			}
+		})
+	}
+}
+
+func benchmarkChunkingGzip(b *testing.B, count int, size int) {
+	mockResponseWriter := httptest.NewRecorder()
+
+	drw := &deferredResponseWriter{
+		mediaType:       "text/plain",
+		statusCode:      200,
+		contentEncoding: "gzip",
+		hw:              mockResponseWriter,
+		ctx:             context.Background(),
+	}
+	for i := 0; i < count; i++ {
+		chunk := []byte(rand2.String(size))
```

Comment:
Benchmarking will include time and allocations needed to randomize input. You need to separate steps for preparing data and measurements.

Reply:
Updated

Reply:
A few seconds later I suddenly realized this